drivers/adt7310/adt7310.c

```diff
@@ -0,0 +1,190 @@
+#define ADT7310_REG_CONFIG (0x01)
+#define ADT7310_REG_VALUE  (0x02)
+#define ADT7310_REG_ID     (0x03)
+#define ADT7310_REG_TCRIT  (0x04)
+#define ADT7310_REG_THYST  (0x05)
+#define ADT7310_REG_THIGH  (0x06)
+#define ADT7310_REG_TLOW   (0x07)
+
+/* ************** *
+ * Register sizes *
+ * ************** */
+#define ADT7310_REG_SIZE_STATUS (1)
+#define ADT7310_REG_SIZE_CONFIG (1)
+#define ADT7310_REG_SIZE_VALUE  (2)
+#define ADT7310_REG_SIZE_ID     (1)
+#define ADT7310_REG_SIZE_TCRIT  (2)
+#define ADT7310_REG_SIZE_THYST  (1)
+#define ADT7310_REG_SIZE_THIGH  (2)
+#define ADT7310_REG_SIZE_TLOW   (2)
+
+/* Register bit masks */
+/** @brief Manufacturer ID */
+#define ADT7310_REG_ID_MASK_MANUFACTURER_ID  (0xF8)
+/** @brief Silicon version */
+#define ADT7310_REG_ID_MASK_SILICON_VERSION  (0x07)
+
+/** @brief Expected manufacturer ID */
+#define ADT7310_EXPECTED_MANUF_ID (0b11000000)
+
+/** @brief Scale factor for temperature readings, floating point number */
+#define ADT7310_TEMPERATURE_LSB (1.f/128.f)
+
+/**
+ * @brief Read a register from the sensor
+ *
+ * @param[in]  dev    device descriptor
+ * @param[in]  addr   register address
+ * @param[in]  len    register size
+ * @param[out] buf    destination buffer
+ */
+static int adt7310_read_reg(const adt7310_t *dev, const uint8_t addr, const uint8_t len,
+                            uint8_t *buf)
+{
+    int status = 0;
+    uint8_t command = ADT7310_CMD_READ | (addr << ADT7310_CMD_ADDR_SHIFT);
+    /* Acquire exclusive access to the bus. */
+    spi_acquire(dev->spi);
+    /* Perform the transaction */
+    gpio_clear(dev->cs);
+
+    if (spi_transfer_regs(dev->spi, command, NULL, (char *)buf, len) < len) {
+        status = -1;
+    }
+
+    gpio_set(dev->cs);
+    /* Release the bus for other threads. */
+    spi_release(dev->spi);
+
+    return status;
+}
+
+/**
+ * @brief Write a register value to the sensor
+ *
+ * @param[in]  dev    device descriptor
+ * @param[in]  addr   register address
+ * @param[in]  len    register size
+ * @param[in]  buf    source buffer
+ */
+static int adt7310_write_reg(const adt7310_t *dev, const uint8_t addr,
+                             const uint8_t len, uint8_t *buf)
+{
+    int status = 0;
+    uint8_t command = ADT7310_CMD_WRITE | (addr << ADT7310_CMD_ADDR_SHIFT);
+    /* Acquire exclusive access to the bus. */
+    spi_acquire(dev->spi);
+    /* Perform the transaction */
+    gpio_clear(dev->cs);
+
+    if (spi_transfer_regs(dev->spi, command, (char *)buf, NULL, len) < len) {
+        status = -1;
+    }
+
+    gpio_set(dev->cs);
+    /* Release the bus for other threads. */
+    spi_release(dev->spi);
+
+    return status;
+}
+
+int adt7310_init(adt7310_t *dev, spi_t spi, gpio_t cs)
+{
+    int status;
+    uint8_t reg = 0;
+    /* write device descriptor */
+    dev->spi = spi;
+    dev->cs = cs;
+    dev->initialized = false;
+
+    /* CS */
+    gpio_init_out(dev->cs, GPIO_NOPULL);
+    gpio_set(dev->cs);
+
+    for (int i = 0; i < 8; ++i) {
```

Comment:
yes, some left over from development, sorry. I've put it inside `#if ENABLE_DEBUG` now.